### DISCUSSION TITLE
Odd Baud Rate

### DISCUSSION THREAD

**Discussion by squareturn** [SpenceKonde/megaTinyCore#710] [Q&A] [answered]
Hi, I'm running the following trivial program on an ATTINY402.  I'm seeing the expected output on pin 2 (TX), but at 66,667 baud.  Am I missing a setup step?
```
#include <Arduino.h>

void
setup() {
    Serial.begin(115200);
    while (!Serial);
}

#define INTERVAL    1000    // ms

void
loop(void) {
    static unsigned long    previous;
    auto                    now = millis();
    
    if (now - previous < INTERVAL) {    // < update ms
        return;
    }
    previous = now;

    Serial.print("UUU ");   // ascii 0x55
    Serial.println(now);
}
```
![Untitled](https://user-images.githubusercontent.com/19697641/168799353-21c6e78f-18e1-452f-9fad-9f8d7190f171.jpg)

**Comment by SpenceKonde** [comment]
That should work if it does not is critical bug

____________
Spence Konde
Azzy’S Electronics

New products! Check them out at tindie.com/stores/DrAzzy
GitHub: github.com/SpenceKonde
ATTinyCore: Arduino support for almost every ATTiny microcontroller
Contact: ***@***.***

**Comment by squareturn** [comment]
I forgot to mention that I'm using megaTinyCore 2.5.11 and Arduino GUI 1.8.19.  Is there any other info you want or debug steps you'd like me to do?

**Comment by hmeijdam** [comment, marked as answer]
And you have tuned the part and not selected the "16MHz internal - tuned"  by accident? If I select the tuned clock on an untuned part I get a random baudrate as well.
If I select just the "16MHz internal" (or any other frequency) the baud on my 404 I used for this test is spot on.

**Comment by squareturn** [comment]
Of course!  That's the problem.

Thank you @hmeijdam I appreciate the help.  Sorry for the false alarm, @SpenceKonde

**Comment by SpenceKonde** [comment]
There is still a bug here.

Selecting a tuned speed without tuning, unless you've written a value to the last 12 bytes of the user row (as you are told not to do if using the tuning option) it should see that that location is blank and take a reasonable guess, whereas it sounds like it is taking a profoundly unreasonable guess of '0'

**Reply by hmeijdam** [reply]
If by "it" you mean the compiler, would it be possible to display a warning in the compiler to remind the user that he forgot to tune the part? Or it will be hard to notice that your assumed tuned value is in fact a not tuned value.
